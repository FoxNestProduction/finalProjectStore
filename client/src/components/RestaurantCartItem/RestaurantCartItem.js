import React from 'react';
import { Box, Typography } from '@mui/material';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import ProductCartItem from '../ProductCartItem/ProductCartItem';
import { totalSumFromCart } from '../Cart/cartFunctions';
import {
  restaurantCartItemContainer,
  textContentWrapper,
  restaurantTitle,
  flexCenter,
  totalWord,
  totalPriceFromRestaurantOrderContainer,
  totalPriceFromRestaurantOrder,
  cartProductsContainer,
} from './styles';

const RestaurantCartItem = ({ restaurantName }) => {
  const restaurants = useSelector((state) => state.cart.restaurants);
  const cartProducts = useSelector((state) => state.cart.cart.products, shallowEqual);
  const filteredProducts = cartProducts.filter((productObj) => (
    productObj.product.restaurant_name === restaurantName
  ));
  const totalSum = totalSumFromCart(filteredProducts);
  console.log(filteredProducts);
  return (
    <Box
      sx={restaurantCartItemContainer}
    >
      <Box
        sx={textContentWrapper}
      >
        <Typography
          variant="h3"
          sx={restaurantTitle}
        >
          {restaurantName}
        </Typography>
        <Box
          sx={flexCenter}
        >
          <Typography
            component="p"
            sx={totalWord}
          >
            Total:
          </Typography>
          <Box
            sx={totalPriceFromRestaurantOrderContainer}
          >
            <Typography
              variant="body3"
              sx={totalPriceFromRestaurantOrder}
            >
              $
              {totalSum}
            </Typography>
          </Box>
        </Box>
      </Box>
      <Box
        sx={cartProductsContainer}
      >
        {filteredProducts.map(({ product, cartQuantity }) => (
          <ProductCartItem key={product._id} cartQuantity={cartQuantity} {...product} />))}
      </Box>
    </Box>
  );
};

RestaurantCartItem.propTypes = {
  restaurantName: PropTypes.string.isRequired,
};

export default RestaurantCartItem;
