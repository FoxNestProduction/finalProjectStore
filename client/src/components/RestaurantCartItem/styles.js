export const flexCenter = {
  display: 'flex',
  alignItems: 'center',
};

export const restaurantCartItemContainer = {
  border: '1px solid',
  borderColor: 'text.primary',
  borderRadius: '18px',
  p: {
    mobile: '10px 5px',
    tablet: '15px 10px',
    lgTablet: '18px 15px 22px',
    desktop: '20px 20px 30px',
  },
};

export const textContentWrapper = {
  ...flexCenter,
  mb: {
    mobile: '10px',
    tablet: '15px',
    lgTablet: '18px',
    desktop: '20px',
  },
  justifyContent: 'space-between',
};

export const restaurantTitle = {
  color: 'text.primary',
  fontSize: {
    mobile: '16px',
    tablet: '18px',
    lgTablet: '24px',
    desktop: '30px',
  },
  mr: {
    mobile: '10px',
  },
};

export const totalWord = {
  color: 'primary.main',
  fontFamily: 'fontPoppins',
  fontWeight: 'fontWeightSemiBold',
  fontSize: {
    mobile: '16px',
    tablet: '18px',
    lgTablet: '22px',
    desktop: '24px',
  },
  lineHeight: {
    mobile: '1.5em',
  },
  paddingRight: {
    mobile: '6px',
    tablet: '8px',
    desktop: '10px',
  },
};

export const totalPriceFromRestaurantOrderContainer = {
  borderRadius: '16px',
  width: {
    mobile: '80px',
    tablet: '85px',
    lgTablet: '100px',
    desktop: '116px',
  },
  backgroundColor: 'primary.main',
  padding: {
    mobile: '4px 0px',
    tablet: '5px 0px',
    lgTablet: '7px 0px',
    desktop: '9px 0px',
  },
  textAlign: 'center',
};

export const totalPriceFromRestaurantOrder = {
  color: 'text.primaryLight',
  fontWeight: 'fontWeightSemiBold',
  fontSize: {
    mobile: '16px',
    tablet: '18px',
    lgTablet: '20px',
    desktop: '22px',
  },
};

export const cartProductsContainer = {
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'stretch',
  gap: {
    mobile: '8px',
    tablet: '15px',
    lgTablet: '18px',
    desktop: '22px',
  },
};
