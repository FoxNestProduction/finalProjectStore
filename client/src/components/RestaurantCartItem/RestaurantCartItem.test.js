import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import RestaurantCartItem from './RestaurantCartItem';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe('RestaurantCartItem Component', () => {

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should renders RestaurantCartItem component', () => {
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce([]);

    const { asFragment } = render(
      <MemoryRouter>
          <RestaurantCartItem restaurantName={'Pesto'} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});