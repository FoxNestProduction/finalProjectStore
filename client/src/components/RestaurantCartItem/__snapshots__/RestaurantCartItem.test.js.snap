// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCartItem Component should renders RestaurantCartItem component 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-dx88nq"
  >
    <div
      class="MuiBox-root css-1b4yrg4"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1sp1o0y-MuiTypography-root"
      >
        Pesto
      </h3>
      <div
        class="MuiBox-root css-70qvj9"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-8bjktx-MuiTypography-root"
        >
          Total:
        </p>
        <div
          class="MuiBox-root css-idyf5l"
        >
          <span
            class="MuiTypography-root MuiTypography-body3 css-ak10t6-MuiTypography-root"
          >
            $0
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1wfkfls"
    />
  </div>
</DocumentFragment>
`;
